domain-models-runtime/src/main/java/org/folio/rest/impl/TenantAPI.java

```diff
@@ -42,70 +44,50 @@
   private static Map<String, List<Promise<Void>>> waiters = new HashMap<>();
 
   PostgresClient postgresClient(Context context) {
     return PostgresClient.getInstance(context.owner());
   }
 
-  Future<Boolean> tenantExists(Context context, String tenantId) {
-    return Future.future(promise -> tenantExists(context, tenantId, promise));
-  }
-
-  void tenantExists(Context context, String tenantId, Handler<AsyncResult<Boolean>> handler){
+  Future<Boolean> tenantExists(Context context, String tenantId){
     /* connect as user in postgres-conf.json file (super user) - so that all commands will be available */
-    postgresClient(context).select(
-      "SELECT EXISTS(SELECT 1 FROM pg_namespace WHERE nspname = '"+ PostgresClient.convertToPsqlStandard(tenantId) +"');",
-        reply -> {
+    return postgresClient(context).select(
+        "SELECT EXISTS(SELECT 1 FROM pg_namespace WHERE nspname = '"
+            + PostgresClient.convertToPsqlStandard(tenantId) +"');")
+        .compose(result -> {
           try {
-            if(reply.succeeded()){
-              Boolean aBoolean = reply.result().iterator().next().getBoolean(0);
-              handler.handle(io.vertx.core.Future.succeededFuture(aBoolean));
-            }
-            else {
-              log.error(reply.cause().getMessage(), reply.cause());
-              handler.handle(io.vertx.core.Future.failedFuture(reply.cause().getMessage()));
-            }
+            Boolean aBoolean = result.iterator().next().getBoolean(0);
+            return Future.succeededFuture(aBoolean);
           } catch (Exception e) {
             log.error(e.getMessage(), e);
-            handler.handle(io.vertx.core.Future.failedFuture(e.getMessage()));
+            return Future.failedFuture(e.getMessage());
           }
-    });
+        });
   }
 
   /**
    * @return previous Schema from rmb_internal.jsonb->>'schemaJson', or null if not exist.
    */
   Future<Schema> previousSchema(Context context, String tenantId, boolean tenantExists) {
-    Promise<Schema> promise = Promise.promise();
-
     if (! tenantExists) {
-      promise.complete(null);
-      return promise.future();
+      return Future.succeededFuture(null);
     }
-
     String sql = "SELECT jsonb->>'schemaJson' " +
         "FROM " + PostgresClient.convertToPsqlStandard(tenantId) + ".rmb_internal";
-    postgresClient(context).selectSingle(sql, select -> {
-      if (select.failed()) {
-        promise.fail(select.cause());
-        return;
-      }
+    return postgresClient(context).selectSingle(sql).compose(row -> {
       try {
-        Row row = select.result();
         String schemaString = row == null ? null : row.getString(0);
         if (schemaString == null) {
-          promise.complete(null);
-          return;
+          return Future.succeededFuture(null);
         }
         Schema schema = ObjectMapperTool.getMapper().readValue(schemaString, Schema.class);
```

Comment:
Creating ObjectMapperTool.readValue(String, Class<T>) that catches IOException and throws it as UncheckedIOException will enable to replace .compose by .map and use .onFailure instead of try-catch.

Reply:
ok. Done